games/devtest/mods/unittests/misc.lua

```diff
@@ -1,15 +1,41 @@
-local function test_random()
+local function test_pseudo_random()
 	-- We have comprehensive unit tests in C++, this is just to make sure the API code isn't messing up
-	local pr = PseudoRandom(13)
-	assert(pr:next() == 22290)
-	assert(pr:next() == 13854)
+	local gen1 = PseudoRandom(13)
+	assert(gen1:next() == 22290)
+	assert(gen1:next() == 13854)
 
-	local pr2 = PseudoRandom(-101)
-	assert(pr2:next(0, 100) == 35)
+	for n = 2, 128 do
+		gen1:next()
+	end
+
+	local gen2 = PseudoRandom(gen1:get_state())
+
+	for n = 128, 256 do
+		assert(gen1:next() == gen2:next())
+	end
```

Comment:
I would reduce this a bit
```suggestion
	local gen2 = PseudoRandom(gen1:get_state())
	for n = 1, 10 do
		assert(gen1:next() == gen2:next())
	end
```